roles/matrix-riot-web/defaults/main.yml

```diff
@@ -28,3 +28,6 @@ matrix_riot_web_embedded_pages_home_path: ~
 
 # Controls whether the self-check feature should validate SSL certificates.
 matrix_riot_web_self_check_validate_certificates: true
+
+# cross use of variable defined in matrix-synapse role
+matrix_riot_web_registration_enabled: "{{ matrix_synapse_enable_registration }}"
```

Comment:
This still references a variable from another role (`matrix-synapse`) right from within the `matrix-riot-web` role.

The way we do this same thing for other roles is:

- we put a sane default value here (`matrix_riot_web_registration_enabled: false`), which doesn't reference any external variables which may not be defined. People who only use the `matrix-riot-web` role (and not the `matrix-synapse` role) outside of this playbook can still have the role function correctly, because each role can be used in isolation.

- do the same cross-referencing in another file, specific to the playbook. This is the `group_vars/matrix-servers` file for us. By adding `matrix_riot_web_registration_enabled: "{{ matrix_synapse_enable_registration }}"` to the **riot-web** section of that file, we wire things the way we want for the playbook and override the sane default value defined in the role.